### PR TITLE
FR-14228 - Step up for Vue 3

### PR DESCRIPTION
**Pull request by doregg**
- Step up for Vue 3
- Fix a bug related to the fallback route on hosted - we do silent refresh and it's unnecessary when the active-url is oauth/callback. Consulted with David.

### CODE REVIEW

File: packages/vue/package.json [hunk 1]
```diff
@@ -22,7 +22,7 @@
     "build:unpkg": "cross-env NODE_ENV=production rollup --config rollup.config.js --format iife"
   },
   "dependencies": {
-    "@frontegg/js": "6.165.0",
+    "@frontegg/js": "6.167.0",
```
Comment (doregg) [inline packages/vue/package.json:25]: merge js update before releasing and clean it

File: packages/vue/src/auth/mapAuthState.ts [hunk 2]
```diff
@@ -217,7 +239,9 @@ export const useFronteggAuthGuard = (options?: FronteggAuthGuardOptions) => {
         if (redirectUrl) {
           localStorage.setItem(FRONTEGG_AFTER_AUTH_REDIRECT_URL, redirectUrl);
         }
-        fronteggAuth.loginActions.requestHostedLoginAuthorize();
+
+        // first time is not a good naming. Here we pass true to check if the active url is oauth/callback to avoid silent refresh and fix a step up bug related to Fallback route
+        fronteggAuth.loginActions.requestHostedLoginAuthorizeV2({firstTime: true, shouldRedirectToLogin: true});
```
Comment (yuvalotem1) [inline packages/vue/src/auth/mapAuthState.ts:244]: We should test this fix carefully because it will load all the initial app requests again

Comment (yuvalotem1) [reply packages/vue/src/auth/mapAuthState.ts:244]: Also `shouldRedirectToLogin: true` can introduce breaking change?

File: packages/vue/src/auth/mapAuthState.ts [hunk 3]
```diff
@@ -156,6 +157,25 @@ export const useLoadEntitlements = () => {
   return inject(loadEntitlementsKey);
 };
 
+/**
+  @param options.maxAge max age in seconds
```
Comment (yuvalotem1) [inline packages/vue/src/auth/mapAuthState.ts:161]:
```suggestion
  @param options.maxAge time limit for step-up session in seconds
```

File: packages/vue/src/auth/mapAuthState.ts [hunk 4]
```diff
@@ -156,6 +157,25 @@ export const useLoadEntitlements = () => {
...
+*/
+export const useIsSteppedUp = (options?: IsSteppedUpOptions) => {
+  return computed(() => {
+    const { user } = inject(authStateKey);
+    return isSteppedUp(user, options);
+  });
+};
+
+/**
+  @param options.maxAge max age in seconds
```
Comment (yuvalotem1) [inline packages/vue/src/auth/mapAuthState.ts:172]: also here explaining maxAge with the words max age is not sufficient

**Review by yuvalotem1 — APPROVED** [review]

File: packages/vue/src/auth/mapAuthState.ts [hunk 5]
```diff
@@ -156,6 +157,25 @@ export const useLoadEntitlements = () => {
   return inject(loadEntitlementsKey);
 };
 
+/**
+  @param options.maxAge max age in seconds
+  @returns whether the user is stepped up
+*/
+export const useIsSteppedUp = (options?: IsSteppedUpOptions) => {
+  return computed(() => {
+    const { user } = inject(authStateKey);
```
[the remark this answers is not in the document]
Comment (doregg) [reply packages/vue/src/auth/mapAuthState.ts:166]: Why do u think that it throws? I must use the inject

**Pull request closed without merging**